src/storage/src/hummock/sstable_store.rs

```diff
@@ -541,66 +403,66 @@ impl SstableStore {
                     Ok(data) => data,
                     Err(e) => {
                         tracing::error!(
-                            "get_block_response meet error when read {:?} from sst-{}, total length: {}",
-                            range,
-                            object_id,
-                            file_size
-                        );
-                        return Err(HummockError::from(e));
+                        "get_block_response meet error when read {:?} from sst-{}, total length: {}",
+                        range,
+                        object_id,
+                        file_size
+                    );
+                        return Err(anyhow::Error::from(HummockError::from(e)));
                     }
                 };
-                let block = Box::new(Block::decode(block_data, uncompressed_capacity)?);
-
-                Ok(block)
+                let block = Box::new(
+                    Block::decode(block_data, uncompressed_capacity)
+                        .map_err(anyhow::Error::from)?,
+                );
+                Ok((block, context))
             }
         };
 
-        let disable_cache: fn() -> bool = || {
-            fail_point!("disable_block_cache", |_| true);
-            false
-        };
-
-        let policy = if disable_cache() {
-            CachePolicy::Disable
-        } else {
-            policy
-        };
-
         if let Some(filter) = self.recent_filter.as_ref() {
             filter.extend([(object_id, usize::MAX), (object_id, block_index)]);
         }
 
         match policy {
-            CachePolicy::Fill(priority) => Ok(self.block_cache.get_or_insert_with(
-                object_id,
-                block_index as u64,
-                priority,
-                fetch_block,
-            )),
-            CachePolicy::FillFileCache => {
-                let block = fetch_block().await?;
-                self.data_file_cache.insert_async(
+            CachePolicy::Fill(context) => {
```

Comment:
just the question: "Is the difference between 'entry' and 'get' related to the requirement for cache population?"

Reply:
`get` only performs get from in-memory cache and the disk cache.

`entry` preforms get if in-memory cache or the disk cache hits, and performs insert and get on cache miss.